src/rt-app_parse_config.c

```diff
@@ -937,6 +1010,7 @@ parse_task_data(char *name, struct json_object *obj, int index,
 			log_info(PIN "Parsing phase %s", key);
 			assure_type_is(val, phases_obj, key, json_type_object);
 			parse_task_phase_data(val, &data->phases[idx], data, opts);
+			check_taskgroup_policy_dep(&data->phases[idx], data);
```

Comment:
a comment saying that you use curr_taskgroup_data and curr_sched_data while parsing the json file to make sure that we don't have unsupported state (policy != other and taskgroup setting) will make sure that we will not break this state machine in the furture

Reply:
I added:

```diff --git a/src/rt-app_parse_config.c b/src/rt-app_parse_config.c
index 18bc62cb9aeb..b286589e2a77 100644
--- a/src/rt-app_parse_config.c
+++ b/src/rt-app_parse_config.c
@@ -986,7 +986,7 @@ parse_task_data(char *name, struct json_object *obj, int index,
        data->sched_data = parse_sched_data(obj, opts->policy);
 
        /*
-        * Thread's current sched_data/taskgroup are used to detect
+        * Thread's current sched_data and taskgroup_data are used to detect
         * policy/taskgroup misconfiguration.
         */
        data->curr_sched_data = data->sched_data;
@@ -1029,6 +1029,10 @@ parse_task_data(char *name, struct json_object *obj, int index,
                        log_info(PIN "Parsing phase %s", key);
                        assure_type_is(val, phases_obj, key, json_type_object);
                        parse_task_phase_data(val, &data->phases[idx], data, opts);
+                       /*
+                        * Uses thread's current sched_data and taskgroup_data
+                        * to detect policy/taskgroup misconfiguration.
+                        */
                        check_taskgroup_policy_dep(&data->phases[idx], data);
                }
 
@@ -1048,6 +1052,10 @@ parse_task_data(char *name, struct json_object *obj, int index,
                free(data->phases[0].sched_data);
                data->phases[0].sched_data = NULL;
 
+               /*
+                * Uses thread's current sched_data and taskgroup_data
+                * to detect policy/taskgroup misconfiguration.
+                */
                check_taskgroup_policy_dep(&data->phases[0], data);
                /*
                 * Get loop number:
@@ -1065,7 +1073,7 @@ parse_task_data(char *name, struct json_object *obj, int index,
                        data->loop = -1;
        }
 
-       /* Reset thread's current sched_data/taskgroup after parsing. */
+       /* Reset thread's current sched_data and taskgroup_data after parsing. */
        data->curr_sched_data = NULL;
        data->curr_taskgroup_data = NULL;
 }
```